rustls/src/common_state.rs

```diff
@@ -682,6 +695,32 @@ impl CommonState {
     }
 }
 
+/// Describes which sort of handshake happened.
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum HandshakeDetails {
+    /// A full handshake.
+    Full,
+
+    /// A full TLS1.3 handshake, with an extra round-trip for a `HelloRetryRequest`.
+    ///
+    /// The server can respond with a `HelloRetryRequest` if the initial `ClientHello`
+    /// is unacceptable for several reasons, the most likely if no supported key
+    /// shares were offered by the client.
+    FullWithHelloRetryRequest,
+
+    /// A resumed handshake.
+    ///
+    /// XXX: deleteme
+    /// In TLS1.2 there are two methods this can happen: via session IDs (with state
+    /// kept by the server) or tickets (with state kept by the client).  It is only
```

Comment:
The "it is only known by the server" part of this seems interesting, but since the information isn't exposed by the variant maybe it shouldn't be mentioned here?

Reply:
Deleted this and replaced it with a more general explanation of what resumption means.